package.json

```diff
@@ -21,21 +21,33 @@
     "clean:build": "echo clean:build",
     "clean:dist": "echo clean:dist",
     "clean:lib": "rimraf ./lib",
+    "commitmsg": "validate-commit-msg",
     "coverage": "yarn test -- --coverage",
-    "format": "eslint --fix src/",
+    "format": "eslint --fix src/js/",
     "generate": "echo generate",
     "lint": "run-s lint:*",
-    "lint:scss": "stylelint --config .stylelintrc.js --syntax scss ./src/scss/**/*.scss",
     "lint:json": "jsonlint-cli {**/*.json,.*rc}",
-    "lint:js": "eslint src/",
-    "commitmsg": "validate-commit-msg",
-    "precommit": "run-s lint test",
+    "lint:js": "eslint src/js/",
+    "lint:scss": "stylelint --syntax scss ./src/scss/**/*.scss",
+    "precommit": "yarn lint-staged && yarn test",
     "prepare": "run-s clean generate build",
     "prepublishOnly": "run-s test",
     "release": "standard-version",
     "test": "jest",
     "watch": "run-p watch:*",
-    "watch:scss": "npm run build:dist:css -- --watch & npm run build:dist:postcss -- --poll=10 -w & npm run build:lib:scss -- --watch"
+    "watch:scss": "yarn build:dist:css -- --watch & yarn build:dist:postcss -- --poll=10 -w & yarn build:lib:scss -- --watch"
+  },
+  "lint-staged": {
+    "*.js": [
+      "eslint --fix",
+      "git add"
+    ],
+    "*.{json,*rc}": [
+      "jsonlint-cli"
+    ],
+    "*.scss": [
+      "stylelint --syntax scss"
```

Comment:
We could potentially use https://github.com/morishitter/stylefmt to fix files before GIT adding them

Reply:
Good idea, but I'm not sure if it supports all our stylelint rules.